src/pivot-table/data/helpers/create-cell.ts

```diff
@@ -1,11 +1,46 @@
 import NxDimCellType from "../../../types/QIX";
 import type { Cell, VisibleDimensionInfo } from "../../../types/types";
+import { MAX_COLUMN_COUNT, MAX_ROW_COUNT } from "../../constants";
 
 // qElemNo === -1 => Total
 // qElemNo === -2 => Null
 // qElemNo === -3 => Others
 // qElemNo === -4 => Empty
 
+const countLeafNodes = (count: number, node: EngineAPI.INxPivotDimensionCell): number =>
+  node.qSubNodes.reduce((acc, childNode) => {
+    if (childNode.qSubNodes.length === 0) {
+      return acc + 1;
+    }
+    return countLeafNodes(acc, childNode);
+  }, count);
+
+const getLeafCount = (
+  node: EngineAPI.INxPivotDimensionCell,
+  y: number,
+  pageY: number,
+  pageX: number,
+  isSnapshot: boolean,
+  isLeftColumn: boolean,
+) => {
+  if (isSnapshot) {
+    return countLeafNodes(0, node);
+  }
+
+  // If a node is at the end of a page and has more child nodes at the next page.
+  // Those child nodes should not me included in the leaf count.
+  const maxLeafCount = isLeftColumn ? MAX_ROW_COUNT - pageY : MAX_COLUMN_COUNT - pageX;
+  let nbrOfLeafNodesThatCanBeFetched = node.qUp + node.qDown;
+  // Leaf count is per PAGE, so if the node is the first node on a page. Ignore any nodes on previous page.
+  if (isLeftColumn && pageY === 0) {
+    const fetchTop = y + node.qUp; // qTop value when node was fetched
+    const nbrOfRowsToFirstRowOnPage = fetchTop % MAX_ROW_COUNT;
+    nbrOfLeafNodesThatCanBeFetched = node.qDown + nbrOfRowsToFirstRowOnPage;
+  }
```

Comment:
shouldn't we just say: 
```
nbrOfLeafNodesThatCanBeFetched = node.qDown
```

Because we start from the beganing of the page? because in that case what matters to us is everything below the node right (`qDown`, probably with a - 1 that subtracts the current node)

Reply:
We are not always starting from the "top" of the page. If a user scrolls down to some y position and expand or collapse a node there. The starting postion is going to be in proximity to where that node was expanded/collapsed.

Which is why `nbrOfRowsToFirstRowOnPage` is needed.

Reply:
but within this if check you have `pageY === 0`, so basically for every page first cell this code will execute and you still calculate `nbrOfRowsToFirstRowOnPage` based on `fetchTop` that uses `y`

or am i missing something maybe?